src/heretic/r_plane.c

```diff
@@ -587,18 +588,20 @@ void R_DrawPlanes(void)
             case 28:
             case 29:           // Scroll_North
                 xsmoothscrolloffset = 0;
-                ysmoothscrolloffset = 0;
+                ysmoothscrolloffset = (crispy->scrolleranims) ? (interpfactor << (pl->special - 25)) -
```

Comment:
There is a lot of duplicate code in this section, which differs only by the value that's substracted from `pl->special`. Could you maybe factor this out into a static function or macro?